### PR TITLE
Fixed quote<->list transformations

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
## Description
This PR aims to fix a bug that I just discovered. Transforming from a quote into a list throws an error, and transforming a list into a quote creates a quote with empty content.
This PR solves both problems and simplifies the logic in a similar way of what was done in paragraph<->list transformations.
Now, transforming from quote to list, creates a list of the quote paragraphs and the last item being the citation. Transforming a list into a quote creates a paragraph for each list item excluding the last one that is used as a citation. This makes the transformations (quote->list->quote, list->quote->list) 
 totally idempotent.

## How Has This Been Tested?
Transform a quote (empty and non-empty) into a list. Verify the first two values of the list are quote, citation. 
Transform a list (empty and non-empty) into a quote. Verify the first two values of the list are now quote and citation of the quote block. If the list contains more than two values just the first two should be used.

### Before:
**List to quote**
nov-16-2017 19-48-38

**Quote to list**
nov-16-2017 19-50-43

### After:
nov-17-2017 11-54-52

### CODE REVIEW

**Review by gziolo — CHANGES_REQUESTED** [review]
Code looks good and works perfectly fine for two items. However, whenever you have more than 2 items on the list or more than two paragraphs in the quote, they are removed when transforming. See screencast:

list-quote

File: blocks/library/list/index.js [hunk 1]
```diff
@@ -162,7 +159,9 @@ registerBlockType( 'core/list', {
 				blocks: [ 'core/quote' ],
 				transform: ( { values } ) => {
 					return createBlock( 'core/quote', {
-						value: [ <p key="list">{ toBrDelimitedContent( values ) }</p> ],
+						value: ( values.length === 1 ? values : initial( values ) )
+							.map( ( value ) => ( { children: <p> { get( value, 'props.children' ) } </p> } ) ),
+						citation: ( values.length === 1 ? undefined : get( last( values ), 'props.children' ) ),
```
Comment (gziolo) [inline blocks/library/list/index.js]: Nice 👍

**Review by gziolo — APPROVED** [review]

**Comment by gziolo** [comment]
It works perfectly fine now. 🙇

**Comment by jorgefilipecosta** [comment]
Thank you for your review @gziolo. 👍

**Pull request merged**